lib/index.js

```diff
@@ -105,8 +106,14 @@ GitHub.prototype.request = function(httpMethod, method, params, opts) {
 
     httpMethod = httpMethod.toUpperCase();
 
-    // Build url
-    uri = url.resolve(this.opts.endpoint, method);
+    // Build url, don't use url.resolve() for github enterprise endpoints:
+    // https://nodejs.org/api/url.html#url_url_resolve_from_to
+    // https://developer.github.com/v3/enterprise/#endpoint-urls
+    if (this.opts.endpoint === DEFAULT_GITHUB_ENDPOINT) {
```

Comment:
Can you use something like [`urljoin`](https://www.npmjs.com/package/urljoin) instead to avoid the condition ?

Here is a code example: https://runkit.com/5845cf2fe3579b001497bbae/5845cf2fe3579b001497bbaf